include/nuttx/mtd/mtd.h

```diff
@@ -116,6 +116,10 @@ struct mtd_geometry_s
   uint32_t erasesize;     /* Size of one erase blocks -- must be a multiple
                            * of blocksize. */
   uint32_t neraseblocks;  /* Number of erase blocks */
+
+  /* NULL-terminated string representing the device model */
+
+  char     model[NAME_MAX + 1];
```

Comment:
It is important that every handler of MTDIOC_GEOMETRY sets the field with a known value (empty string).
This should be part of this PR, so that the application is able to distinguish when the `model` is indeed available.

Reply:
Done.